sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/UriPropertiesShouldNotBeStrings.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+
+namespace Tests.Diagnostics
+{
+    class Foo
+    {
+        string Url { get; set; } // Noncompliant
+//      ^^^^^^
+    }
+
+    class Bar : Foo
+    {
+        override string Url { get; set; } // Compliant
+    }
+
+    class Program
+    {
+        // Url
+        string Url { get; set; } // Noncompliant {{Change this property type to 'System.Uri'}}
```

Comment:
I would use message like `Change the type of the '{property-name}' property to 'System.Uri'.`. Also note that we put full stop at the end of the messages.